mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/bridge/builtin/impl/DefaultIntegerIdentifierBridge.java

```diff
@@ -23,4 +23,9 @@ public Integer fromDocumentIdentifier(String documentIdentifier, IdentifierBridg
 		return Integer.parseInt( documentIdentifier );
 	}
 
+	@Override
+	public boolean isCompatibleWith(IdentifierBridge<?> bridge) {
+		return false;
```

Comment:
Something like `return getClass().equals( bridge.getClass() )` would be more appropriate.